src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,116 @@
 /* eslint-disable */
 import React from 'react';
 
+import { useAppSelector } from '../../app/hooks';
+import { useDispatch } from 'react-redux';
+import { Todo } from '../../types/Todo';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { Status } from '../../types/Status';
+
+import classNames from 'classnames';
+
 export const TodoList: React.FC = () => {
+  const todosFilter = useAppSelector(state => state.todos);
```

Comment:
You are using `useAppSelector` to select the entire `todos` state. It would be more efficient to create and use a selector that filters the todos based on the filter state. This way, the filtering would be done in the Redux store and the component would only re-render when the filtered todos change.